ztest: Add register functionality

yperess opened the pull request:
Add new functionality to ztest to improve test modularity. The two
primary new entry points are:
* ztest_register_test_suite
* ztest_run_registered_test_suites

When registering a new test suite, users provide the name as well as
an optional pragma used to filter the tests for each run. Using NULL
as the pragma ensures that the test is run exactly once (after which
it is removed from the linked list of test suites).

Calls to ztest_run_registered_test_suites take a state pointer as an
argument. This allows the the pragma functions to decide whether the
test should be run (see the BT and accel tests as an example of using
the global test state as a filter).

The biggest benefit of this system (other than the ability to filter
tests and maintain a larger test state) is the ability to better
modularize the test source code. In the example of basicmath, main.c
doesn't need to (or should) know about q7, q15, q31, or f32's test
suites. Instead, each source file manages registering its own test
suite and handling the conditions for running the suite.

Signed-off-by: Yuval Peress <peress@chromium.org>

Commit:
ztest: Add register functionality

Add new functionality to ztest to improve test modularity. The two
primary new entry points are:
* ztest_register_test_suite
* ztest_run_registered_test_suites

When registering a new test suite, users provide the name as well as
an optional predicate used to filter the tests for each run. Using NULL
as the predicate ensures that the test is run exactly once (after which
it is automatically filtered from future runs).

Calls to ztest_run_registered_test_suites take a state pointer as an
argument. This allows the the pragma functions to decide whether the
test should be run.

The biggest benefit of this system (other than the ability to filter
tests and maintain a larger test state) is the ability to better
modularize the test source code. Instead of all the various tests
having to coordinate and the main function having to know which tests
to run, each source file manages registering its own test
suite and handling the conditions for running the suite.

Signed-off-by: Yuval Peress <peress@chromium.org>

stephanosio commented on subsys/testsuite/ztest/src/ztest.c:
```diff
@@ -427,6 +427,41 @@ void end_report(void)
+	while (current_node) {
+		bool should_run = true;
+
+		if (current_node->pragma) {
```
Violation: Rule 14.4
```suggestion
		if (current_node->pragma != NULL) {
```

stephanosio commented on subsys/testsuite/ztest/src/ztest.c:
```diff
@@ -427,6 +427,41 @@ void end_report(void)
+	struct unit_test_node *current_node = z_ztest_unit_test_head;
+	int count = 0;
+
+	while (current_node) {
```
Violation: Rule 14.4
```suggestion
	while (current_node != NULL) {
```

stephanosio commented on subsys/testsuite/ztest/src/ztest.c:
```diff
@@ -427,6 +427,41 @@ void end_report(void)
+			++count;
+		}
+		/* Remove the node if pragma is NULL. This ensures that the suite runs only once. */
+		if (!current_node->pragma) {
```
Violation: Rule 14.4
```suggestion
		if (current_node->pragma == NULL) {
```

stephanosio commented on subsys/testsuite/ztest/src/ztest.c:
```diff
@@ -427,6 +427,41 @@ void end_report(void)
+		}
+		/* Remove the node if pragma is NULL. This ensures that the suite runs only once. */
+		if (!current_node->pragma) {
+			if (prev_node) {
```
Violation: Rule 14.4
```suggestion
			if (prev_node != NULL) {
```

stephanosio commented on subsys/testsuite/ztest/include/ztest_test.h:
```diff
@@ -27,6 +27,63 @@ struct unit_test {
+	static int z_ztest_suite_##SUITE_NAME##_init(const struct device *dev)                     \
+	{                                                                                          \
+		ARG_UNUSED(dev);                                                                   \
+		struct unit_test_node *old_head = z_ztest_unit_test_head;                          \
+		z_ztest_unit_test_head = &z_ztest_test_node_##SUITE_NAME;                          \
+		z_ztest_test_node_##SUITE_NAME.next = old_head;                                    \
```
`old_head` seems to be unnecessary here.

stephanosio reviewed (changes requested):
**Should add tests under `tests/ztest` that exercise these new features.**

Also, at first glance, there seems to be a fatal flaw in the linked list logic implemented here:
_Initial run with current = head, prev = NULL, the first two nodes have NULL pragma_
1. current pragma is NULL and prev is NULL, so head is set to the next node effectively removing it from the next run; prev is now the current (removed) node and current is now the next node.
2. current pragma is NULL and prev was set to the removed node, so prev's next is set to current's next. But prev has already been removed by updating head to the current node and head still points to the node that should have been removed.

Also the CI didn't run `tests/benchmarks/cmsis_dsp/basicmath` and is giving a pass -- compilation fails for this test locally.

yperess commented:
Done with the requested changes and added unit tests under `tests/ztest/register`

yperess commented:
I'm going to need a bit of help with this. After adding the tests under `tests/ztest/register` I'm getting am error running twister:

```
/usr/bin/ld: ../app/libapp.a(main.c.obj): warning: relocation in read-only section `.z_init_APPLICATION99_'
/usr/bin/ld: warning: creating DT_TEXTREL in a PIE
collect2: error: ld returned 1 exit status
gmake[2]: *** [zephyr/CMakeFiles/zephyr_prebuilt.dir/build.make:113: zephyr/zephyr_prebuilt.elf] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:2701: zephyr/CMakeFiles/zephyr_prebuilt.dir/all] Error 2
gmake: *** [Makefile:103: all] Error 2
```

I'm assuming this is caused by me manipulating the linked list at runtime? Is there any way to suppress this for the test?

stephanosio commented on tests/drivers/sensor/accel/src/main.c:
```diff
@@ -55,17 +55,33 @@ static void run_tests_on_accel(const char *label)
 
-#if DT_NODE_EXISTS(DT_ALIAS(accel_1))
-	run_tests_on_accel(DT_LABEL(DT_ALIAS(accel_1)));
-#endif
```
`DT_NODE_EXISTS(DT_ALIAS(accel_1))` is statically computable, so I am not sure what benefit this can offer over the good old `#if` approach -- if any, it makes it harder to understand the overall flow by moving the control statement to elsewhere.

This might make sense in the languages that support lambda expression and the pragma can be declared as a lambda, but that is unfortunately not the case in C -- even then, its usage should be limited to non-statically-computable cases.

stephanosio reviewed (changes requested):
Please separate the updates to the existing tests into their own commits (e.g. `tests/bluetooth/df/connectionless_cte_tx` in one commit and `tests/benchmarks/cmsis_dsp/basicmath` in another, ...) so they can be reverted individually in case we find something wrong with them in the future.

stephanosio commented:
> I'm going to need a bit of help with this. After adding the tests under tests/ztest/register I'm getting am error running twister:

I just tested it locally and it seems to work fine.

```
$ scripts/twister -v -N -T tests/ztest/register
ZEPHYR_BASE unset, using "/home/stephanos/Dev/zephyrproject/zephyr"
Renaming output directory to /home/stephanos/Dev/zephyrproject/zephyr/twister-out.10
INFO    - Zephyr version: zephyr-v2.6.0-1222-g7d25de0ae171
INFO    - JOBS: 60
INFO    - Using 'zephyr' toolchain.
INFO    - Selecting default platforms per test case
INFO    - Building initial testcase list...
INFO    - 1 test scenarios (32 configurations) selected, 13 configurations discarded due to filters.
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    -  1/19 native_posix              tests/ztest/register/testing.ztest.register        PASSED (native 2.156s)
INFO    -  2/19 mps2_an385                tests/ztest/register/testing.ztest.register        PASSED (qemu 2.313s)
INFO    -  3/19 mps2_an521                tests/ztest/register/testing.ztest.register        PASSED (qemu 2.440s)
INFO    -  4/19 qemu_cortex_m0            tests/ztest/register/testing.ztest.register        PASSED (qemu 2.187s)
INFO    -  5/19 nsim_sem                  tests/ztest/register/testing.ztest.register        PASSED (build)
INFO    -  6/19 frdm_k64f                 tests/ztest/register/testing.ztest.register        PASSED (build)
INFO    -  7/19 nsim_hs_smp               tests/ztest/register/testing.ztest.register        PASSED (build)
INFO    -  8/19 qemu_riscv32              tests/ztest/register/testing.ztest.register        PASSED (qemu 2.183s)
INFO    -  9/19 qemu_nios2                tests/ztest/register/testing.ztest.register        PASSED (qemu 2.081s)
INFO    - 10/19 qemu_riscv64              tests/ztest/register/testing.ztest.register        PASSED (qemu 2.062s)
INFO    - 11/19 qemu_xtensa               tests/ztest/register/testing.ztest.register        PASSED (qemu 2.106s)
INFO    - 12/19 qemu_leon3                tests/ztest/register/testing.ztest.register        PASSED (qemu 2.055s)
INFO    - 13/19 qemu_x86                  tests/ztest/register/testing.ztest.register        PASSED (qemu 2.208s)
INFO    - 14/19 qemu_arc_hs               tests/ztest/register/testing.ztest.register        PASSED (qemu 2.112s)
INFO    - 15/19 qemu_arc_em               tests/ztest/register/testing.ztest.register        PASSED (qemu 2.158s)
INFO    - 16/19 qemu_cortex_a53           tests/ztest/register/testing.ztest.register        PASSED (qemu 2.120s)
INFO    - 17/19 qemu_cortex_a53_smp       tests/ztest/register/testing.ztest.register        PASSED (qemu 2.173s)
INFO    - 18/19 qemu_x86_64               tests/ztest/register/testing.ztest.register        PASSED (qemu 2.337s)
INFO    - 19/19 qemu_cortex_r5            tests/ztest/register/testing.ztest.register        PASSED (qemu 8.594s)

INFO    - 19 of 19 test configurations passed (100.00%), 0 failed, 13 skipped with 0 warnings in 27.96 seconds
INFO    - In total 19 test cases were executed, 13 skipped on 19 out of total 380 platforms (5.00%)
INFO    - 16 test configurations executed on platforms, 3 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing xunit report /home/stephanos/Dev/zephyrproject/zephyr/twister-out/twister.xml...
INFO    - Writing xunit report /home/stephanos/Dev/zephyrproject/zephyr/twister-out/twister_report.xml...
INFO    - Run completed
```

Commit:
tests: basicmath: Migrate to new ztest register functionality

Simplify the running and creation of the basicmath benchmark suites.
Using the ztest register functionality, each benchmark registers and
is run from a single call in main.c.

Signed-off-by: Yuval Peress <peress@chromium.org>

Commit:
tests: BT: migrate connectionless tx to ztest register

Migrating the test suites to the register functionality allows the
main.c source to focus on the state of the system instead of "which"
test to run. In this example, test_main() now calls:
* ut_bt_setup() and updates the state
* common_create_adv_set(), updates the state, and notifies the
  registered tests that the state has been updated which would trigger
  a run.
* common_delete_adv_set(), updates the state, and again notifies
  the registered tests that the state has been updated.

Signed-off-by: Yuval Peress <peress@chromium.org>

yperess commented:
Split this PR into 3 commits

stephanosio commented on tests/ztest/register/testcase.yaml:
```diff
@@ -0,0 +1,5 @@
+tests:
+  testing.ztest.register:
+    tags: test_framework
+    integration_platforms:
+    - native_posix
```
While not incorrect, should follow the standard indentation style, which is also used throughout the Zephyr repo.
```suggestion
    integration_platforms:
      - native_posix
```

stephanosio reviewed (commented):
Looks ok overall.

Minor comment about YAML indent style.

stephanosio commented on tests/benchmarks/cmsis_dsp/basicmath/src/q7.c:
```diff
@@ -276,18 +276,11 @@ void test_benchmark_vec_dot_prod_q7(void)
-
-	ztest_run_test_suite(basicmath_q7_benchmark);
-}
+ztest_register_test_suite(basicmath_q7_benchmark, NULL, ztest_unit_test(test_benchmark_vec_add_q7),
+			  ztest_unit_test(test_benchmark_vec_sub_q7),
+			  ztest_unit_test(test_benchmark_vec_mult_q7),
+			  ztest_unit_test(test_benchmark_vec_abs_q7),
+			  ztest_unit_test(test_benchmark_vec_negate_q7),
+			  ztest_unit_test(test_benchmark_vec_offset_q7),
+			  ztest_unit_test(test_benchmark_vec_scale_q7),
+			  ztest_unit_test(test_benchmark_vec_dot_prod_q7));
```
```suggestion
ztest_register_test_suite(basicmath_q7_benchmark, NULL,
			  ztest_unit_test(test_benchmark_vec_add_q7),
			  ztest_unit_test(test_benchmark_vec_sub_q7),
			  ztest_unit_test(test_benchmark_vec_mult_q7),
			  ztest_unit_test(test_benchmark_vec_abs_q7),
			  ztest_unit_test(test_benchmark_vec_negate_q7),
			  ztest_unit_test(test_benchmark_vec_offset_q7),
			  ztest_unit_test(test_benchmark_vec_scale_q7),
			  ztest_unit_test(test_benchmark_vec_dot_prod_q7));
```

stephanosio reviewed (approved):
Looks ok to me.

jackrosenthal commented on subsys/testsuite/ztest/include/ztest_test.h:
```diff
@@ -27,6 +29,62 @@ struct unit_test {
+		z_ztest_unit_test_head = &z_ztest_test_node_##SUITE_NAME;                          \
+		return 0;                                                                          \
+	}                                                                                          \
+	SYS_INIT(z_ztest_suite_##SUITE_NAME##_init, APPLICATION, 99)
```
can we use a linker section instead of init+linked list?

yperess replied:
Sure I was planning on doing this in another PR to keep this one a bit simpler, but here's what I was thinking. @stephanosio could you take a look and let me know what you think:

1. Each node will also hold a `struct test_suite_stats` which will count `num_runs`, `num_failed`, and `num_skipped`.
2. Nodes will no longer be removed, but instead skipped if `num_runs != 0 && pragma == NULL`
3. Introduce a new macro `ztest_num_tests_not_run()` which will count the number of tests that have `num_runs == 0`.
4. Introduce `ztest_verify_all_tests_ran()` which will add a new test that simply checks that no node exists with `num_runs == 0`. This can be added at the end of `test_main()` to make sure that all the test suites actually did run.
5. Introduce `CONFIG_ZTEST_MAIN` that will add the following code to `ztest.c`:
```
#ifdef CONFIG_ZTEST_MAIN
void test_main(void)
{
  ztest_run_registered_test_suites();
  ztest_verify_all_tests_ran();
}
#endif /* CONFIG_ZTEST_MAIN */
```

I could also just add the `num_runs` component of the first bullet and implement the second in this PR, then leave the rest to the following. Please let me know if you would like me to leave things as they are, make minimal changes to this PR, or make all the changes in this PR.

nashif commented on tests/benchmarks/cmsis_dsp/basicmath/src/f32.c:
```diff
@@ -348,18 +348,12 @@ void test_benchmark_vec_dot_prod_f32(void)
 
-void benchmark_basicmath_f32(void)
-{
-	ztest_test_suite(basicmath_f32_benchmark,
```
we have some very primitive scanning in twister to identify testcases, see https://github.com/zephyrproject-rtos/zephyr/blob/718c77a4cdc4aa0e3b31e51feb15f1d7638b211f/scripts/pylib/twister/twisterlib.py#L1653 for example. This needs to be adapted to also parse ztest_register_test_suite

yperess replied:
Thanks for catching this. I had to get a bit creative since the call to register the test suite and the call to run it may not be in the same file and the call to run the suites doesn't specify which tests to run. I effectively looked across all the files to see if tests were registered and if so, required at least one call to run them.

yperess replied:
I'll add unit tests to twister shortly, just wanted to upload the current setup as early as I could to make sure everyone is happy with the approach.

yperess replied:
OK, tests were updated and everything seems to be passing on my end.

yperess commented:
@stephanosio I made changes to how this all works based on Jack and Anas' requests. Would you be able to take another quick look to make sure this still looks good to you?

stephanosio commented on subsys/testsuite/ztest/include/ztest_test.h:
```diff
@@ -27,7 +29,65 @@ struct unit_test {
+	uint32_t fail_count;
+};
+
+/**
+ * A single node of test suite. Each node should be added to unit_test_head to
+ * form a linked list that can be run via ztest_run_registered_test_suites().
+ */
```
The comment should be updated since that is no longer the case.

stephanosio commented on subsys/testsuite/ztest/include/ztest_test.h:
```diff
@@ -27,7 +29,65 @@ struct unit_test {
+extern const struct ztest_suite_node __ztest_suite_end[];
+
+/**
+ * Create and register a ztest suite. Using this macro creates a new test suite (using
+ * ztest_test_suite). It then creates a struct unit_test_node and an initializer function which will
+ * be run via a SYS_INIT call at the APPLICATION initialization level and priority 99.
```
This needs to be updated as well.

stephanosio commented on subsys/testsuite/ztest/include/ztest_test.h:
```diff
@@ -27,7 +29,65 @@ struct unit_test {
+#define ztest_register_test_suite(SUITE_NAME, PRAGMA, ...)                                         \
+	ztest_test_suite(SUITE_NAME, __VA_ARGS__);                                                 \
+	static struct ztest_suite_node z_ztest_test_node_##SUITE_NAME                              \
+		__attribute__((__section__(".ztest_suites"))) __used = {                           \
```
Should use an abstraction like `Z_GENERIC_SECTION` rather than gcc-specific attributes.

stephanosio commented on subsys/testsuite/ztest/src/ztest.c:
```diff
@@ -427,6 +429,37 @@ void end_report(void)
 K_APPMEM_PARTITION_DEFINE(ztest_mem_partition);
 #endif
 
+struct unit_test_node *z_ztest_unit_test_head;
```
No longer needed.

stephanosio commented on subsys/testsuite/ztest/src/ztest.c:
```diff
@@ -427,6 +429,37 @@ void end_report(void)
+			int fail = z_ztest_run_test_suite(ptr->name, ptr->suite);
+
+			count++;
+			ptr->metadata.run_count++;
+			ptr->metadata.fail_count += fail ? 1 : 0;
```
Here we modify `metadata` which is a member of `const`-qualified/rodata(ROM)-allocated data. We cannot do this.

stephanosio commented on subsys/testsuite/ztest/include/ztest_test.h:
```diff
@@ -27,7 +29,65 @@ struct unit_test {
+	struct ztest_suite_metadata metadata;
+};
+
+extern const struct ztest_suite_node __ztest_suite_start[];
+extern const struct ztest_suite_node __ztest_suite_end[];
```
Must remove `const` since `metadaa`, which resides inside `ztest_suite_node` is mutable.

stephanosio commented on include/linker/common-rom.ld:
```diff
@@ -155,6 +155,15 @@
 	} GROUP_LINK_IN(ROMABLE_REGION)
 #endif /* CONFIG_EMUL */
 
+#if defined(CONFIG_ZTEST)
+	SECTION_DATA_PROLOGUE(ztest_section,,)
+	{
+		__ztest_suite_start = .;
+		KEEP(*(SORT_BY_NAME(".ztest_suites")));
+		__ztest_suite_end = .;
+	} GROUP_LINK_IN(ROMABLE_REGION)
+#endif /* CONFIG_ZTEST */
```
Since `metadata`, which resides in this section, is mutable, we must make this a RAM section.

yperess replied:
Thanks for pointing me in the right direction, can't say I'm a wiz on linker scripts :)

Commit:
ztest: add a weak implementation of test_main()

Introduce a weak implementation of test_main() which calls:
* ztest_run_registered_test_suites(NULL);
* ztest_verify_all_registered_test_suites_ran();

This will attempt to run all registered test suites and verify that
they each ran.

Signed-off-by: Yuval Peress <peress@chromium.org>

yperess commented:
I'm not sure why https://buildkite.com/zephyr/zephyr/builds/37040 is failing, I'm able to build these (I tried a couple) locally.

yperess commented:
So the issue is the unit tests that specify `unit_testing` as the board ends up not using the `common-ram.ld`. I'm not able to find which linker files are used in those cases though. Any help would be appreciated.

yperess commented:
Got it, all checks are now passing.

stephanosio commented on subsys/testsuite/include/ztest.ld:
```diff
@@ -0,0 +1,16 @@
+SECTIONS
+{
+	.data.ztest_suite_node_area : ALIGN(4)
+  	{
```
Mixing SPACEs and TABs here. Should stick with TABs.

yperess replied:
Thanks, snuck past me :)

stephanosio reviewed (commented):
Looks ok overall.

would like to get some feedback from @nashif for the twister-related changes.

yperess commented:
Ping

yperess commented:
@nashif Would you have some cycles to take a look at this?

nashif commented:
> @nashif Would you have some cycles to take a look at this?

yes, on my todo..

yperess commented:
@jackrosenthal Any chance you might have some cycles to take another pass at this one? I got everything working like you asked with the linker section.

sjg20 reviewed (changes requested):
It sems like there should be a documentation update included here somewhere

Commit:
doc: ztest: Update documentation for register

Add documentation that details the ability to register test suites and
execute them based on a pragma function. Also the auto-generated
test_main() function.

Signed-off-by: Yuval Peress <peress@chromium.org>

yperess commented:
I've added the docs, but I'm getting a warning that I'm not seeing locally about the new macro being deprecated:

```
/workdir/zephyr/tests/ztest/automain/src/test_feature.c:20:13: error: Macro is deprecated [-Werror]
   20 | ztest_register_test_suite(test_suite, NULL, ztest_unit_test(test_stub));
      |             ^~~~~~~~~~~~~~~~~~~~~
```

Any idea what's different about the CI?

yperess commented on scripts/tests/twister/test_testinstance.py:
```diff
@@ -13,7 +13,8 @@
 sys.path.insert(0, os.path.join(ZEPHYR_BASE, "scripts/pylib/twister"))
-from twisterlib import TestInstance, BuildError, TestCase, TwisterException
+from twisterlib import TestInstance, BuildError, TestCase, TwisterException, \
+    ScanPathResult
```
Done

yperess commented on tests/ztest/register/include/common.h:
```diff
@@ -0,0 +1,23 @@
+	PHASE_STEPS_0,
+	PHASE_STEPS_1,
+};
+struct state {
```
Done

yperess commented:
Ping on this PR, do we have enough approvals to merge? I understand that the 2.7 release branch has been cut which means we're merging into main now?

yperess commented:
@sjg20 @nashif it's still showing that there were requested changes from both of you, if you can, could you PTAL and let me know if anything is missing? This should make it a lot easier to script system level / integration tests.

yperess commented:
This PR has had 3 approvals for quite some time now, can it please merge? @nashif can you help with this? I'm really starting to pile on tech debt.

yperess commented:
Force push to re-trigger timed out documentation build

tejlmand commented on tests/ztest/register/CMakeLists.txt:
```diff
@@ -0,0 +1,10 @@
+# Copyright 2021 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.13.1)
```
should be updated to `3.20.0`
```suggestion
cmake_minimum_required(VERSION 3.20.0)
```

yperess replied:
Done

tejlmand commented on tests/ztest/automain/CMakeLists.txt:
```diff
@@ -0,0 +1,9 @@
+# Copyright 2021 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.13.1)
```
```suggestion
cmake_minimum_required(VERSION 3.20.0)
```

tejlmand reviewed (changes requested):
Minimum CMake is now 3.20.0

yperess commented on tests/bluetooth/df/connectionless_cte_tx/src/main.c:
```diff
@@ -7,17 +7,25 @@
-#include "test_set_cl_cte_tx_enable.h"
+
+struct bt_test_state test_state = {
+	.is_setup = false,
```
Done.

yperess commented on tests/bluetooth/df/connectionless_cte_tx/src/common.h:
```diff
@@ -8,6 +8,12 @@ extern struct bt_le_ext_adv *g_adv;
 extern uint8_t g_cte_len;
 
+struct bt_test_state {
+	bool is_setup;
```
Done

yperess commented on tests/bluetooth/df/connectionless_cte_tx/src/test_set_cl_cte_tx_param.c:
```diff
@@ -223,19 +223,23 @@ void test_set_ct_cte_tx_params_aoa_invalid_pattern_len(void)
 }
 
-void run_set_cl_cte_tx_params_tests(void)
+static bool test_set_cl_cte_tx_param_pragma(const void *state)
```
Done

yperess commented:
> I notice that @sjg20 has approved, but could you please reply to his comments so that we know why they are not addressed. For example: [#37192 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/37192#discussion_r696835037) [#37192 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/37192#discussion_r696840106) [#37192 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/37192#discussion_r696841225)

Missed those, fixed them.
